packages/integration-tests/src/test-project-setup/test_project_base.ts

```diff
@@ -144,4 +149,40 @@ export abstract class TestProjectBase {
 
     assert.ok(clientConfigStats.isFile());
   }
+
+  /**
+   * Verify deployed client outputs
+   */
+  async assertDeployedClientOutputs(backendId: BackendIdentifier) {
+    const { BackendOutputClientFactory: npmBackendOutputClientFactory } =
+      await import(
+        pathToFileURL(
+          path.join(
+            this.projectDirPath,
+            'node_modules',
+            '@aws-amplify',
+            'deployed-backend-client',
+            'lib',
+            'backend_output_client_factory.js'
+          )
+        ).toString()
+      );
+
+    const currentCodebaseBackendOutputClient =
+      CurrentCodebaseBackendOutputClientFactory.getInstance({
+        getAmplifyClient: () => this.amplifyClient,
+        getCloudFormationClient: () => this.cfnClient,
+      });
+
+    const npmBackendOutputClient = npmBackendOutputClientFactory.getInstance({
+      getAmplifyClient: () => this.amplifyClient,
+      getCloudFormationClient: () => this.cfnClient,
+    });
+
+    const currentCodebaseOutputs =
+      await currentCodebaseBackendOutputClient.getOutput(backendId);
+    const npmOutputs = await npmBackendOutputClient.getOutput(backendId);
+
+    assert.deepStrictEqual(currentCodebaseOutputs, npmOutputs);
+  }
```

Comment:
What if there are intentional non-breaking changes being launched? I'm guessing this will fail all e2e deployment tests as soon as this condition is hit preventing further checks to be run?

Reply:
Yes this will still fail when there are intentional changes (breaking and non-breaking). In the case of this assert failing, the only checks that are not run are for subsequent test projects for branch deployment.

I was thinking this assertion will be like the API check so PR reviewers can see the changes being made from the backend client's perspective.